presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/AddExchanges.java

```diff
@@ -288,104 +282,34 @@ public PlanWithProperties visitAggregation(AggregationNode node, Context context
             }
 
             if (node.getGroupBy().isEmpty()) {
-                if (decomposable) {
-                    return splitAggregation(node, child, partial -> gatheringExchange(idAllocator.getNextId(), REMOTE, partial));
-                }
-                else {
-                    child = withDerivedProperties(
-                            gatheringExchange(idAllocator.getNextId(), REMOTE, child.getNode()),
-                            child.getProperties());
+                child = withDerivedProperties(
+                        gatheringExchange(idAllocator.getNextId(), REMOTE, child.getNode()),
+                        child.getProperties());
 
-                    return rebaseAndDeriveProperties(node, child);
-                }
+                return rebaseAndDeriveProperties(node, child);
             }
             else {
                 if (child.getProperties().isStreamPartitionedOn(partitioningRequirement)) {
                     return rebaseAndDeriveProperties(node, child);
                 }
                 else {
                     if (decomposable) {
```

Comment:
now that decomposing is in another optimizer, we shouldn't even need to be checking for this in this optimizer, so we can simplify this a bit.